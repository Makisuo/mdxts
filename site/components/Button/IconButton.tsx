'use client'
import * as React from 'react'
import { Box } from 'components/Box'
import { ButtonProps, variants } from 'components/Button'

/**
 * IconButton allows users to take specific actions using a simplified icon graphic.
 */
export function IconButton({
  icon,
  variant = 'primary',
  backgroundColor: propsBackgroundColor,
  color: propsColor,
  style,
  ...props
}: Omit<ButtonProps, 'children'> & { icon: string }) {
  const { backgroundColor, color } = variants[variant] || {}

  return (
    <Box
      as="button"
      {...props}
      style={{
        ...style,
        appearance: 'none',
        backgroundColor: propsBackgroundColor || backgroundColor,
        color: propsColor || color,
      }}
    >
      {icon}
    </Box>
  )
}
