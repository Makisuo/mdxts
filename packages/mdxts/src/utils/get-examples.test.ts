import { Project } from 'ts-morph'
import {
  getExamplesFromDirectory,
  getExamplesFromExtension,
  getExamplesFromSourceFile,
} from './get-examples'

const buttonSourceFile = `
/** Used for taking actions and navigating. */
export function Button({
  label,
  onPress,
}: {
  label: string
  onPress: () => void
}) {
  return <button onClick={onPress}>{label}</button>
}
`

const basicUsageSourceFile = `
import { Button } from '../Button'

export function BasicUsage() {
  return <Button label="Say hello" onPress={() => alert('Hello!')} />
}

export function AlternateUsage() {
  return <Button label="Say goodbye" onPress={() => alert('Goodbye!')} />
}
`

test('get examples from directory', async () => {
  const project = new Project({ useInMemoryFileSystem: true })
  const directory = project.createDirectory('directory')

  const sourceFile = directory.createSourceFile('Button.tsx', buttonSourceFile)

  directory
    .createDirectory('examples')
    .createSourceFile('BasicUsage.tsx', basicUsageSourceFile)

  const examples = getExamplesFromDirectory(sourceFile.getDirectory())

  expect(examples.map((example) => example.getFilePath())).toMatchSnapshot()
})

test('get examples from extension', async () => {
  const project = new Project({ useInMemoryFileSystem: true })
  const sourceFile = project.createSourceFile('Button.tsx', buttonSourceFile)

  project.createSourceFile('Button.examples.tsx', basicUsageSourceFile)

  const examples = getExamplesFromExtension(sourceFile)

  expect(examples!.getFilePath()).toMatchSnapshot()
})

test('get examples from source file', async () => {
  const project = new Project({ useInMemoryFileSystem: true })
  const sourceFile = project.createSourceFile('Button.tsx', buttonSourceFile)

  project.createSourceFile('Button.examples.tsx', basicUsageSourceFile)

  const examples = await getExamplesFromSourceFile(sourceFile, {
    '/Button.examples.tsx': Promise.resolve({
      BasicUsage: () => null,
      AlternateUsage: () => null,
    }),
  })

  expect(examples).toMatchSnapshot()
})
