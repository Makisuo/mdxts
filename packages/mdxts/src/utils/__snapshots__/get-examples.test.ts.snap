// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get examples from directory 1`] = `
[
  "/directory/examples/BasicUsage.tsx",
]
`;

exports[`get examples from extension 1`] = `"/Button.examples.tsx"`;

exports[`get examples from source file 1`] = `
[
  {
    "moduleExport": [Function],
    "name": "BasicUsage",
    "slug": "basic-usage",
    "sourcePath": "",
    "sourceText": "import { Button } from '../Button'

export function BasicUsage() {
  return <Button label="Say hello" onPress={() => alert('Hello!')} />
}",
  },
  {
    "moduleExport": [Function],
    "name": "AlternateUsage",
    "slug": "alternate-usage",
    "sourcePath": "",
    "sourceText": "import { Button } from '../Button'

export function AlternateUsage() {
  return <Button label="Say goodbye" onPress={() => alert('Goodbye!')} />
}",
  },
]
`;
